nemo/collections/asr/models/slu_models.py

```diff
@@ -0,0 +1,325 @@
+# ! /usr/bin/python
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Dict, Optional
+from urllib.error import HTTPError
+
+import torch
+from omegaconf import DictConfig, open_dict
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.data.audio_to_text_dali import DALIOutputs
+from nemo.collections.asr.losses.slu_losses import SmoothedNLLLoss
+from nemo.collections.asr.models.ctc_bpe_models import EncDecCTCModelBPE
+from nemo.collections.asr.parts.utils.slu_utils import SearcherConfig, SequenceGenerator, get_seq_mask
+from nemo.collections.common.parts.adapter_modules import LinearAdapterConfig
+from nemo.core import adapter_mixins
+from nemo.core.classes.common import Serialization, typecheck
+from nemo.core.neural_types import AudioSignal, LabelsType, LengthsType, LogprobsType, NeuralType, SpectrogramType
+from nemo.utils import logging
+
+
+class SLUIntentSlotBPEModel(EncDecCTCModelBPE):
+    def __init__(self, cfg: DictConfig, trainer=None):
+        if hasattr(cfg, "adapter") and getattr(cfg.adapter, "enabled", False):
```

Comment:
Umm this isnt necessary. By default we assume class does *not* support adapters, if it did, then it must save model with correct config anyway, so this test and check is not necessary.